src/mpyl/steps/models.py

```diff
@@ -107,7 +107,7 @@ def from_configuration(run_properties: Dict, config: Dict):
 
         return RunProperties(
             details=RunContext.from_configuration(build['run']),
-            target=Target(build['parameters']['deploy_target']),
+            target=Target(build['parameters']['deploy_target'] or Target.PULL_REQUEST.value),
```

Comment:
This feels a bit hacky but it's probably just my limited python knowledge. It's creating a new `Target` from the value of an already existing one. Plus the `or` only covers the scenario in which 'deploy_target' is `None` right? But we made it optional above, so it could also not exist at all and give a KeyError anyway.